### PR TITLE
PERF: Improve performance for pyarrow to_numpy with na and na_value

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

saw this in the engine pr, avoids a warning and improves performance:

```
pr
%timeit ser.to_numpy(dtype="int64", na_value=1.5)
1.67 ms ± 11.2 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

main
%timeit ser.to_numpy(dtype="int64", na_value=1.5)
/Users/patrick/PycharmProjects/pandas/pandas/core/arrays/arrow/array.py:893: RuntimeWarning: invalid value encountered in cast
  result = np.asarray(self._data, dtype=dtype)
2.42 ms ± 30.3 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

### CODE REVIEW

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -889,9 +889,13 @@ def to_numpy(
             result = np.empty(len(self), dtype=object)
             mask = ~self.isna()
             result[mask] = np.asarray(self[mask]._data)
+        elif self._hasna:
+            data = self.copy()
+            data[self.isna()] = na_value
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:894]: So we can't necessarily do `self._data.fill_null(na_value)`?

Comment (phofl) [reply pandas/core/arrays/arrow/array.py:894]: We'd have to check if the na_value is compatible with the arrow array. This raises if not, right now we fall back to object. Is there a function for this in arrow?

Comment (mroeschke) [reply pandas/core/arrays/arrow/array.py:894]: That's what I thought. Yeah there isn't one AFAICT so this makes sense.

Comment (phofl) [reply pandas/core/arrays/arrow/array.py:894]: We could try except it? Not really elegant :)

Would cut runtime in half again, so would be a nice gain

**Comment by phofl** [comment]
greenish

**Pull request merged**